### PR TITLE
Remove mInfo field from Cancelable by default.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
It's big, and totally unused.

The opt-in to enable it is for backwards compat.  If we care about preserving compat while also ensuring that some internal Cancelable uses never have this field, that is also possible, but more complicated.

### CODE REVIEW

File: src/lib/core/core.gni [hunk 1]
```diff
@@ -84,6 +84,10 @@ declare_args() {
 
   # When using minmdns, set the number of parallel resolves
   chip_config_minmdns_max_parallel_resolves = 2
+
+  # If set to true, adds a string "info" field to Cancelable.
+  # Only here for backwards compat.  Generally should not be set to true.
```
Comment (tcarmelveilleux) [inline src/lib/core/core.gni:89]:
```suggestion
  # Only here for backwards compat.  Generally, THIS SHOULD NOT BE SET TO TRUE.
```

**Review by tcarmelveilleux — APPROVED** [review]

**Comment by gmarcosb** [comment]
👏🎉 that one snuck in there didn't it 😅

**Comment by mspang** [comment]
Do we need the backward compat? I think it could have been removed in #14581

**Comment by bzbarsky-apple** [comment]
> Do we need the backward compat?

@mspang it's technically public API at this point... unlike when https://github.com/project-chip/connectedhomeip/pull/14581 merged, which was before we shipped, right?

**Comment by mspang** [comment]
> > Do we need the backward compat?
> 
> @mspang it's technically public API at this point... unlike when #14581 merged, which was before we shipped, right?

I suspect even if somebody spotted this storage they would not have risked stashing anything there.

Keeping an option seems safe, but I would suggest we aim to remove it later since it likely does not impact anyone.

**Review by mspang — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]

**Pull request merged**